airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/csv_parser.py

```diff
@@ -47,78 +52,134 @@ async def infer_schema(
+                + config_format.skip_rows_after_header
+            )
+            self._skip_rows(fp, rows_to_skip)
+
+            reader = csv.DictReader(fp, dialect=dialect_name, fieldnames=headers)  # type: ignore
+            try:
+                for row in reader:
+                    # The row was not properly parsed if any of the values are None. This will most likely occur if there are more columns
+                    # than headers or more headers dans columns
+                    if None in row or None in row.values():
+                        raise RecordParseError(FileBasedSourceError.ERROR_PARSING_RECORD)
+                    yield row
+            finally:
+                # due to RecordParseError or GeneratorExit
+                csv.unregister_dialect(dialect_name)
+
+    def _get_headers(self, fp: IOBase, config_format: CsvFormat, dialect_name: str) -> List[str]:
+        # Note that this method assumes the dialect has already been registered if we're parsing the headers
+        self._skip_rows(fp, config_format.skip_rows_before_header)
+        if config_format.autogenerate_column_names:
+            return self._auto_generate_headers(fp, config_format, dialect_name)
+        else:
+            # Then read the header
+            reader = csv.reader(fp, dialect=dialect_name)  # type: ignore
+            return list(next(reader))
+
+    def _auto_generate_headers(self, fp: IOBase, config_format: CsvFormat, dialect_name: str) -> List[str]:
+        """
+        Generates field names as [f0, f1, ...] in the same way as pyarrow's csv reader with autogenerate_column_names=True.
+        See https://arrow.apache.org/docs/python/generated/pyarrow.csv.ReadOptions.html
+        """
+        reader = csv.reader(fp, dialect=dialect_name)  # type: ignore
+        number_of_columns = len(next(reader))  # type: ignore
+        # Reset the file pointer to the beginning of the file so that the first row is not skipped
+        fp.seek(0)
+        return [f"f{i}" for i in range(number_of_columns)]
+
+    @staticmethod
+    def _skip_rows(fp: IOBase, rows_to_skip: int) -> None:
+        """
+        Skip rows before the header. This has to be done on the file object itself, not the reader
+        """
+        for _ in range(rows_to_skip):
+            fp.readline()
+
+
+class CsvParser(FileTypeParser):
+    _MAX_BYTES_PER_FILE_FOR_SCHEMA_INFERENCE = 1_000_000
+
+    def __init__(self, csv_reader: Optional[_CsvReader] = None):
+        self._csv_reader = csv_reader if csv_reader else _CsvReader()
+
+    async def infer_schema(
+        self,
+        config: FileBasedStreamConfig,
+        file: RemoteFile,
+        stream_reader: AbstractFileBasedStreamReader,
+        logger: logging.Logger,
+    ) -> SchemaType:
+        if config.input_schema:
+            # FIXME change type of method to Mapping
+            return config.input_schema  # type: ignore # conversion to mapping is handled by pydantic and we shouldn't have a str here
```

Comment:
to follow up, it will always be a string, but given the FileBasedStreamConfig, you can get the mapping representation by using the `get_input_schema()` method which returns an optional